### ISSUE TITLE
CVE-2022-22947 not detected

### ISSUE THREAD

**Issue by peter-cui1221** [aquasecurity/trivy#1983]
## Checklist
- [ Y ] I've read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/latest/community/contribute/issue/#wrong-detection).
- [ Y ] I've confirmed that a security advisory in data sources was correct.
    - Run Trivy with `-f json` that shows data sources and make sure that the security advisory is correct.

## Description

In spring cloud gateway versions prior to 3.1.1+ and 3.0.7+ , applications are vulnerable to a code injection attack when the Gateway Actuator endpoint is enabled, exposed and unsecured. A remote attacker could make a maliciously crafted request that could allow arbitrary remote execution on the remote host.

## JSON Output of run with `-debug`:

```
trivy image vulhub/spring-cloud-gateway:3.1.0 
2022-04-13T10:26:48.870+0800	INFO	Detected OS: debian
2022-04-13T10:26:48.870+0800	INFO	Detecting Debian vulnerabilities...
2022-04-13T10:26:48.886+0800	INFO	Number of language-specific files: 1
2022-04-13T10:26:48.886+0800	INFO	Detecting jar vulnerabilities...

vulhub/spring-cloud-gateway:3.1.0 (debian 10.9)
===============================================
Total: 244 (UNKNOWN: 1, LOW: 134, MEDIUM: 29, HIGH: 59, CRITICAL: 21)
Java (jar)
==========
Total: 6 (UNKNOWN: 2, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 3)

+---------------------------------------------+------------------+----------+-------------------+--------------------+---------------------------------------+
|                   LIBRARY                   | VULNERABILITY ID | SEVERITY | INSTALLED VERSION |   FIXED VERSION    |                 TITLE                 |
+---------------------------------------------+------------------+----------+-------------------+--------------------+---------------------------------------+
| com.fasterxml.jackson.core:jackson-databind | CVE-2020-36518   | HIGH     | 2.13.1            | 2.12.6.1, 2.13.2.1 | jackson-databind: denial of service   |
| (spring-cloud-gateway-0.0.1-SNAPSHOT.jar)   |                  |          |                   |                    | via a large depth of nested objects   |
|                                             |                  |          |                   |                    | -->avd.aquasec.com/nvd/cve-2020-36518 |
+---------------------------------------------+------------------+----------+-------------------+--------------------+---------------------------------------+
| org.springframework.boot:spring-boot        | CVE-2022-22965   | CRITICAL | 2.6.4             | 2.5.12, 2.6.6      | spring-framework: RCE via             |
| (spring-cloud-gateway-0.0.1-SNAPSHOT.jar)   |                  |          |                   |                    | Data Binding on JDK 9+                |
|                                             |                  |          |                   |                    | -->avd.aquasec.com/nvd/cve-2022-22965 |
+                                             +------------------+----------+                   +                    +---------------------------------------+
|                                             | GMS-2022-560     | UNKNOWN  |                   |                    | Improper Neutralization of            |
|                                             |                  |          |                   |                    | Special Elements used in an OS        |
|                                             |                  |          |                   |                    | Command ('OS Command...               |
+                                             +------------------+          +                   +                    +                                       +
|                                             | GMS-2022-561     |          |                   |                    |                                       |
|                                             |                  |          |                   |                    |                                       |
|                                             |                  |          |                   |                    |                                       |
+---------------------------------------------+------------------+----------+-------------------+--------------------+---------------------------------------+
| org.springframework:spring-beans            | CVE-2022-22965   | CRITICAL | 5.3.16            | 5.3.18, 5.2.20     | spring-framework: RCE via             |
| (spring-cloud-gateway-0.0.1-SNAPSHOT.jar)   |                  |          |                   |                    | Data Binding on JDK 9+                |
|                                             |                  |          |                   |                    | -->avd.aquasec.com/nvd/cve-2022-22965 |
+---------------------------------------------+                  +          +                   +                    +                                       +
| org.springframework:spring-webflux          |                  |          |                   |                    |                                       |
| (spring-cloud-gateway-0.0.1-SNAPSHOT.jar)   |                  |          |                   |                    |                                       |
|                                             |                  |          |                   |                    |                                       |
+---------------------------------------------+------------------+----------+-------------------+--------------------+---------------------------------------+
```

## Output of `trivy -v`:

```
wanlxp@B-488YMD6R-2001 trivy % trivy -v
Version: 0.25.4
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-04-13 00:10:11.199700745 +0000 UTC
  NextUpdate: 2022-04-13 06:10:11.199700145 +0000 UTC
  DownloadedAt: 2022-04-13 02:08:32.388711 +0000 UTC
```

## Additional details (base image name, container registry info...):
Image name: vulhub/spring-cloud-gateway:3.1.0

**Comment by DmitriyLewen** [comment]
Hello @peter-cui1221 
Thanks for your report!

I checked your image.
I found `spring-cloud-gateway` package, but this package has `org.vulhub` prefix:
```
➜  issue1983 trivy image -f json --list-all-pkgs vulhub/spring-cloud-gateway:3.1.0 | grep -E '"Name.*spring-cloud-gateway'
          "Name": "org.springframework.cloud:spring-cloud-gateway-server",
          "Name": "org.vulhub:spring-cloud-gateway",
```

 `MANIFEST.MF` and `pom.properties` files from `spring-cloud-gateway-0.0.1-SNAPSHOT.jar` also have this prefix. 

[Github Database](https://github.com/advisories/GHSA-3gx9-37ww-9qw6) and [GitLab Database](https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/blob/master/maven/org.springframework.cloud/spring-cloud-gateway/CVE-2022-22947.yml#L6) work with `org.springframework.cloud:spring-cloud-gateway` package.

That is why Trivy doesn't detect `CVE-2022-22947` in your image.

Regards, Dmitriy
